backend/services/music_gen_service/music_tools.py

```
@@ -0,0 +1,1561 @@
+            raise ValueError(f"Invalid note name: {note}")
+            
+        # Calculate MIDI pitch
+        pitch = c_value + offset
+        
+        # Validate pitch is in MIDI range
+        if pitch < 0 or pitch > 127:
+            raise ValueError(f"Pitch out of MIDI range (0-127): {pitch}")
+            
+        return pitch
+    
+    def _convert_duration_to_beats(self, duration_str, time_signature: List[int] = [4, 4]) -> float:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)